### PR TITLE
Use resource filtering to update vaadin version

### PR DESCRIPTION
**Pull request by pleku**
Makes it possible to build and ship the archetypes with two maven commands.

### CODE REVIEW

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -22,6 +21,7 @@
         <repository>
             <id>Vaadin Directory</id>
             <url>http://maven.vaadin.com/vaadin-addons</url>
+            <snapshots>false</snapshots>
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: Does this work like this too? I have always use this:

            <snapshots>
                <enabled>false</enabled>
            </snapshots>

Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: It don't, tested. 
```suggestion
            <snapshots><enabled>false</enabled></snapshots>
```

Comment (ZheSun88) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: yeah. exactly the problem.

Comment (ZheSun88) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: I pushed a commit there ;-) i hope @pleku doesn't mind.

Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: You could have just pushed Commit suggestion button too ;-)

**Comment by ZheSun88** [comment]
build failed on cannot find the `vaadin-bom 13.0-SNAPSHOT` dependency. 
an easy fix would be adding the snapshot repository, 
but it looks we don't use snapshot repository in V8 version either.

**Comment by ZheSun88** [comment]
Add snapshot repository. 

In FW8, the [goal](https://github.com/vaadin/archetypes/blob/V8/vaadin-archetype-application/src/test/resources/projects/basic/goal.txt) file uses a profile in the command line which contains the snapshot repository settings.

**Comment by mstahv** [comment]
I don't think we need pre-releases of archetypes at least at this point. Maybe later when we want advanced users to try 14 pre-releases.

vaadin.version property should default to some valid released version. You can then override that with build configurations, but I actually wouldn't suggest to do that (as builds should be repeatable).

File: pom.xml [hunk 2]
```diff
@@ -11,10 +11,12 @@
     </parent>
     <groupId>com.vaadin</groupId>
     <artifactId>vaadin-archetypes</artifactId>
-    <version>13.0.2</version>
+    <version>13.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
-
+    <properties>
+        <vaadin.version>${project.version}</vaadin.version>
```
Comment (mstahv) [inline pom.xml]:
```suggestion
        <vaadin.version>13.0.2</vaadin.version>
```

Project version is not the same as vaadin version used by outputted project. If we build an archetype with SNAPSHOT vaadin version, we should have snapshot repository declared.

Comment (ZheSun88) [reply pom.xml]: updated the `vaadin.version`; 
removed the snapshot repository, as it is not needed if we use a release platform version

File: pom.xml [hunk 3]
```diff
@@ -11,10 +11,12 @@
     </parent>
     <groupId>com.vaadin</groupId>
     <artifactId>vaadin-archetypes</artifactId>
-    <version>13.0.2</version>
+    <version>13.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
-
+    <properties>
+        <vaadin.version>13.0.2</vaadin.version>
```
Comment (pleku) [inline pom.xml]: So now the release build has to also use `versions:update-properties` in addition to `versions:set`.

**Pull request merged**